climada/hazard/base.py

```diff
@@ -369,6 +371,104 @@ def set_vector(self, *args, **kwargs):
                        "Use Hazard.from_vector instead.")
         self.__dict__ = Hazard.from_vector(*args, **kwargs).__dict__
 
+    @classmethod
+    def from_raster_netcdf(
+        cls,
+        data: Union[xr.Dataset, str],
+        *,
+        intensity: str = "intensity",
+        fraction: Union[str, Callable[[np.ndarray], np.ndarray]] = "fraction",
+        coordinate_vars: Optional[Dict[str, str]] = None,
+    ):
+        """Read raster-like data from a NetCDF file
+
+        Parameters
+        ----------
+        data : xarray.Dataset or str
+            The NetCDF data to read from. May be a opened dataset or a path to a NetCDF
+            file, in which case the file is opened first.
+        intensity : str
+            Identifier of the `xarray.DataArray` containing the hazard intensity data.
+        fraction : str or Callable
```

Comment:
I think it would be simpler to have `str` with default `None`. If `None`, the fraction is set to one. I do not really see a use case with another callable right now?

Reply:
It would be more useful if the callable could also take the coordinates as arguments, but I feel like this complicates matters too much. With the current implementation, you can use the callable, e.g., for implementing a simple function based on the intensity, like a custom threshold. It would be more simple if we left it out, but it also feels like a rather low-hanging fruit. What's the harm in leaving it in?

Reply:
I removed the option to insert a callable and made `fraction` an optional variable that is set via `data_vars`.

Reply:
Let's see how the use of this method evolves, and see whether we come back on this decision later or not. I like it this way, as it starts with the simpler version.